packages/block-library/src/image/edit.js

```diff
@@ -278,27 +321,139 @@ class ImageEdit extends Component {
 	}
 
 	updateAlignment( nextAlign ) {
-		const extraUpdatedAttributes = [ 'wide', 'full' ].indexOf( nextAlign ) !== -1 ?
-			{ width: undefined, height: undefined } :
-			{};
-		this.props.setAttributes( { ...extraUpdatedAttributes, align: nextAlign } );
+		if ( nextAlign === 'wide' || nextAlign === 'full' ) {
+			// Reset all sizing attributes.
+			this.resetWidthHeight();
+		}
+
+		this.props.setAttributes( { align: nextAlign } );
+	}
+
+	updateImageURL( url, imageSizeOptions ) {
```

Comment:
What is the expected shape of the second argument here? It seems like it could one or either of an object or an array of said objects? The overloading is hard to follow. For the array form, it also seems like the function could just call to `this.getImageSizeOptions` directly, which would leave open the option for the second argument to serve the sole purpose of being an explicit width/height.

Reply:
> It seems like it could one or either of an object or an array of said objects

Yes, it reuses the array passed to the image sizes drop-down. That was the "original" use. Added support for the object with width and height so it can be reused when updating the URL of uploaded image. Of course can have another function there, or just add that inline.

Reply:
I'd love to see a second function instead of overloading parameter types. it took me considerable time to try and figure out what was happening here, especially since this PR only calls it with one of the types.